database/src/models.rs

```diff
@@ -290,6 +289,19 @@ impl Account {
             .optional()
     }
 
+    pub fn fetch_by_username_domain(
+        db_conn: &Connection,
+        username: impl Into<String>,
+        domain: Option<impl Into<String>>,
+    ) -> QueryResult<Option<Account>> {
+        use super::schema::accounts::dsl;
+        dsl::accounts
+            .filter(dsl::username.eq(username.into()))
+            .filter(dsl::domain.is_not_distinct_from(domain.map(Into::into)))
```

Comment:
Curious, why `is_not_distinct_from` and not `eq`? (Does it have something to do with `NULL` in the domain column?)

The truth table for `IS NOT DISTINCT FROM` is more complex, so this piqued my curiosity.  (Also asking because AFAIK it's not supported directly by SQLite, which is something I'm still poking at.)

If the function of `IS NOT DISTINCT FROM` is required, `a IS NOT DISTINCT FROM b` has equivalents, as shown e.g. at https://wiki.postgresql.org/wiki/Is_distinct_from.

Reply:
oh, yes, this was a bit of a slapdash hack around the `NULL`s.
turns out `NULL` is not `==` to `NULL` in postgres, so doing `.eq(None)` won't work. to get around this, i used `IS DISTINCT FROM`. the best way to remove this would be using a [boxed query](https://github.com/diesel-rs/diesel/issues/455) to do something like (pseudocode obvs)
```rs
if let Some(domain) = domain.into() {
    .eq(domain)
} else {
    .is_null()